src/_pytest/config/__init__.py

```diff
@@ -1305,6 +1316,69 @@ def _getconftest_pathlist(
             values.append(relroot)
         return values
 
+    def _get_append_ini_values(self, name: str, ini_type) -> List[str]:
+        value = []
+        # append_ini is a list of "ini=value" options
+        # append all values if multiple values are set for same ini-name,
+        # e.g. -a foo=bar1 -a foo=bar2 will append 'bar1 bar2' to foo
+        for ini_config in self._append_ini:
+            try:
+                key, append_ini_value = ini_config.split("=", 1)
+            except ValueError:
+                raise UsageError(
+                    "-a/--append-ini expects option=value style (got: {!r}).".format(
+                        ini_config
+                    )
+                )
+            else:
+                if key == name:
+                    if ini_type in ["pathlist", "args"]:
+                        value.append(append_ini_value)
+                    elif ini_type == "linelist":
+                        value.append("\n{}".format(append_ini_value))
+                    else:
+                        self._warn_or_fail_if_strict(
+                            "append_ini option invalid for argument '{}' with type '{}'".format(
+                                name, ini_type
+                            )
+                        )
+        return value
+
+    def _resolve_ini_option(
+        self,
+        name: str,
+        default_value: str,
+        override_value: Optional[str],
+        append_values: List,
+        ini_type: Optional[str],
+    ) -> Union[List, str, None]:
+        value = self.inicfg.get(name, None)
+        if override_value is None and not append_values:
+            if value is None:
+                if default_value is not None:
+                    return default_value
+                if ini_type is None:
+                    return ""
+                return []
+        else:
+            if override_value:
+                value = override_value
+            if append_values:
+                if not value:
```

Comment:
What is the reason you chose to fail in this case instead of just "appending to empty"?

Reply:
I don't think it makes sense to append to nothing. If we do that we're saying append to undefined which I think deserves a fail.

It would also be a little bit more confusing ( at least to me ) to have that happen from a workflow perspective. Users could create a garbage ini key and append to it without defining it. That could cause a good amount of headache for workflows

Reply:
Can you describe what you mean by "Users could create a garbage ini key"?

In any case let me describe why I think appending should work:

Let's say I want to change the `norecursedirs` ini. Let's split to 4 cases:

1. I want to completely reset `norecursedirs` and set it to my own -- I'll use `--override-ini`.
2. I want to add a value in addition to what may or may not exists in my config, and there exists values in the config.
3. I want to add a value in addition to what may or may not exists in my config, and there doesn't exists values in the config.
4. I want to add a value, but only if there doesn't exist values in my config.

Case 1 is pretty clear - use `--override-ini`.

Case 4 is something I can't really see anyone wanting.

Cases 2 & 3 is a difference I don't really want to think about; I move between different projects and I don't want to remember or check whether the config already defines a value (then I should use `--append-ini`) or doesn't (then I should use `--override-ini`). I think it's better to always allow to use `--append-ini`.

Reply:
I meant that the user could assume that since the key was appended to, it must already be defined somewhere, so perhaps it would open up room for confusion? Now that I've read points 2 and 3 I see your point more clearly and agree. I'll make the change